File: projects/lab4/task5/app/src/main/java/com/example/lab3_rework/MainActivity.java
package com.example.lab3_rework;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.NavigationUI;

import com.google.android.material.navigation.NavigationView;


public class MainActivity extends AppCompatActivity{

    NavController navContr;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        NavigationView view = findViewById(R.id.drawer_nav_view);
        navContr = Navigation.findNavController(this, R.id.nav_host);
        NavigationUI.setupActionBarWithNavController(this, navContr);
        NavigationUI.setupWithNavController(view, navContr);
    }

    @Override
    public boolean onSupportNavigateUp() {
        return navContr.navigateUp();
    }

}

File: projects/lab6/lab7_task3/app/src/main/java/com/example/lab7_3/MainActivity.kt
package com.example.lab7_3

import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity


class MainActivity : AppCompatActivity() {

    private lateinit var imageView: ImageView
    private lateinit var button: Button
    private val myData = ViewModelImage()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        imageView = findViewById(R.id.image)
        button = findViewById(R.id.button)

        button.setOnClickListener {
            if (myData.getBitmapMy() == null) {
                myData.down()
                Log.d(TAG, "image downloaded")
            }
        }

        myData.mutableLiveData.observe(this) {
            imageView.setImageBitmap(myData.getBitmapMy())
            Log.d(TAG, "set image")
        }
    }

    companion object {
        const val TAG = "MyIm"
    }
}